src/zcash/IncrementalMerkleTree.cpp

```diff
@@ -26,27 +38,41 @@ SHA256Compress SHA256Compress::combine(const SHA256Compress& a, const SHA256Comp
 }
 
 
-template <typename Hash>
+template <size_t Depth, typename Hash>
 class PathFiller {
 private:
     std::deque<Hash> queue;
 public:
     PathFiller() : queue() { }
     PathFiller(std::deque<Hash> queue) : queue(queue) { }
 
-    Hash next() {
+    Hash next(size_t depth) {
         if (queue.size() > 0) {
             Hash h = queue.front();
             queue.pop_front();
 
             return h;
         } else {
-            return Hash();
+            return IncrementalMerkleTree<Depth, Hash>::rootOfEmptyTree(depth);
         }
     }
+
 };
 
 template<size_t Depth, typename Hash>
+Hash IncrementalMerkleTree<Depth, Hash>::rootOfEmptyTree(size_t depth) {
+    boost::optional<boost::array<Hash, Depth+1>> empty_roots;
+    if (!empty_roots) {
```

Comment:
I'm confused by this, it looks as though it is trying to cache the empty roots, but then ``empty_roots`` is declared as local.